roles/ups/tasks/main.yml

```diff
@@ -42,4 +49,18 @@
 - name: Delete ups temporary file
   file:
     path: "/tmp/unifiedpushserver.yml"
-    state: absent
\ No newline at end of file
+    state: absent
+
+- name: "Wait for UPS DB pod to be ready"
+  shell: "oc get pods --namespace={{ mobile_services_namespace }} --selector=deploymentconfig=unifiedpush-postgresql -o jsonpath='{.items[*].status.phase}' | grep Running"
+  register: result
+  until: result.rc == 0
+  retries: 50
+  delay: 10
+
+- name: "Wait for UPS pod to be ready"
+  shell: "oc get pods --namespace={{ mobile_services_namespace }} --selector=deploymentconfig=unifiedpush -o jsonpath='{.items[*].status.phase}' | grep Running"
```

Comment:
Hi @wei-lee I noticed in the case of ups that the status of the deployment will be Running but the pod itself doesn't come up for a while after that. In that case the script moves on and deploys the other resources. I've deleted more projects and minishift doesn't crash but it does struggle while all the other services and ups are coming. It does come up in the end so it's probably fine but just wanted to note it.

Reply:
yes, we just check if the pod is running, and didn't check if the container is ready. The problem with checking if container is ready is that there could be multiple containers running, and if we want to check all of them we may have to hard code the number of containers here, which will make this hard to maintain. In most cases, just checking the pod is running should be enough.